cms/djangoapps/contentstore/views/component.py

```diff
@@ -140,6 +141,9 @@ def container_handler(request, usage_key_string):
             is_unit_page = is_unit(xblock)
             unit = xblock if is_unit_page else None
 
+            if use_new_unit_page(usage_key.course_key):
```

Comment:
```suggestion
            if is_unit_page and use_new_unit_page(usage_key.course_key):
```
`unit` needs a check for None here, which is kind of solved by adding the condition.